rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java

```
@@ -107,38 +109,55 @@ public SelectionManager(RMCore rmcore) {
             File folder = new File(dirName);
+
             if (folder.exists() && folder.isDirectory()) {
-                authorizedSelectionScripts = new HashSet<>();
-                for (File file : folder.listFiles()) {
-                    if (file.isFile()) {
-                        try {
-                            String script = SelectionScript.readFile(file);
-                            logger.debug("Adding authorized selection script " + file.getAbsolutePath());
-                            authorizedSelectionScripts.add(SelectionScript.digest(script));
-                        } catch (Exception e) {
-                            logger.error(e.getMessage(), e);
-                        }
-                    }
+                logger.info("The resource manager will accept only selection scripts from " + dirName);
+                long currentTime = System.currentTimeMillis();
+                long configuredAuthorizedScriptLoadPeriod = getConfiguredAuthorizedScriptLoadPeriod();
+                if (currentTime - lastAuthorizedFolderLoadingTime > configuredAuthorizedScriptLoadPeriod) {
+                    lastAuthorizedFolderLoadingTime = currentTime;
+                    loadAuthorizedScriptsSignatures(folder);
                 }
             } else {
                 logger.error("Invalid dir name for authorized scripts " + dirName);
             }
         }
     }
 
+    private void loadAuthorizedScriptsSignatures(File folder) {
+        authorizedSelectionScripts = new HashSet<>();
+        for (File file : folder.listFiles()) {
+            if (file.isFile()) {
+                try {
+                    String script = SelectionScript.readFile(file);
+                    logger.debug("Adding authorized selection script " + file.getAbsolutePath());
+                    authorizedSelectionScripts.add(Script.digest(script.trim()));
+                } catch (Exception e) {
+                    logger.error(e.getMessage(), e);
```

Here we can throw an exception